instrumentation/http-tests/src/main/java/brave/test/http/ITHttpServer.java

```diff
@@ -386,30 +390,82 @@ public void addsStatusCode_badRequest() throws Exception {
       .containsEntry("error", "400");
   }
 
+  /**
+   * Some synchronous frameworks have limited means to adjust the HTTP status code upon raising an
+   * exception. When this is the case, use the following built-in exception:
+   *
+   * <p><pre>{@code
+   *   throw new UnavailableException("not ready", 1); // implies 503
+   * }</pre>
+   */
+  @Test
+  public void httpStatusCodeTagMatchesResponse_onException() throws Exception {
+    httpStatusCodeTagMatchesResponse_onException("/exception");
+  }
+
+  @Test
+  public void httpStatusCodeTagMatchesResponse_onException_async() throws Exception {
+    httpStatusCodeTagMatchesResponse_onException("/exceptionAsync");
+  }
+
+  Span httpStatusCodeTagMatchesResponse_onException(String path) throws Exception {
+    Response response = get(path);
+
+    Span span = takeSpan();
+    assertThat(span.tags())
+      .containsEntry("http.status_code", String.valueOf(response.code()));
+
+    return span;
+  }
+
   @Test
-  public void reportsSpanOnException() throws Exception {
-    reportsSpanOnException("/exception");
+  public void errorTag_exceptionOverridesHttpStatus() throws Exception {
+    errorTag_exceptionOverridesHttpStatus("/exception");
   }
 
   @Test
-  public void reportsSpanOnException_async() throws Exception {
-    reportsSpanOnException("/exceptionAsync");
+  public void errorTag_exceptionOverridesHttpStatus_async() throws Exception {
+    errorTag_exceptionOverridesHttpStatus("/exceptionAsync");
   }
 
-  Span reportsSpanOnException(String path) throws Exception {
+  void errorTag_exceptionOverridesHttpStatus(String path) throws Exception {
     get(path);
 
-    return takeSpan();
+    Span span = takeSpan();
+    assertThat(span.tags().get("error"))
+      .contains("not ready"); // some controllers format the exception
```

Comment:
when I made the assertion actually look at content, I found most of our code didn't return what we thought it would.. rather it just set the http status.